lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+  } = require('kuzzle-common-objects').errors;
 
 /**
+ * @class Validation
  * @param {Kuzzle} kuzzle
- * @constructor
  */
-function Validation (kuzzle) {
-  /** @type {Kuzzle} */
-  this.kuzzle = kuzzle;
+class Validation {
+  constructor(kuzzle) {
+    /** @type {Kuzzle} */
+    this.kuzzle = kuzzle;
 
-  /** @type {...ValidationType} */
-  this.types = {};
+    /** @type {...ValidationType} */
+    this.types = {};
 
-  /** @type {string[]} */
-  this.typeAllowsChildren = [];
+    /** @type {string[]} */
+    this.typeAllowsChildren = [];
 
-  /** @type {DocumentSpecification} */
-  this.specification = {};
+    /** @type {DocumentSpecification} */
+    this.specification = {};
 
-  /** @type {Dsl} */
-  this.dsl = new Dsl();
+    /** @type {Dsl} */
+    this.dsl = new Dsl();
 
-  this.rawConfiguration = {};
-}
+    this.rawConfiguration = {};
+  }
 
-/**
- * Walks through all types in "defaultTypesFiles" initializes all types
- */
-Validation.prototype.init = function validationInit () {
-  var ValidationType;
-
-  [
-    'anything',
-    'boolean',
-    'date',
-    'email',
-    'enum',
-    'geoPoint',
-    'geoShape',
-    'integer',
-    'ipAddress',
-    'numeric',
-    'object',
-    'string',
-    'url'
-  ].forEach(typeFile => {
-    ValidationType = require(`./types/${typeFile}`);
-    this.addType(new ValidationType());
-  });
-};
+  /**
+   * Walks through all types in "defaultTypesFiles" initializes all types
+   */
+  init() {
+    [
+      'anything',
+      'boolean',
+      'date',
+      'email',
+      'enum',
+      'geoPoint',
+      'geoShape',
+      'integer',
+      'ipAddress',
+      'numeric',
+      'object',
+      'string',
+      'url'
+    ].forEach(typeFile => {
+      const TypeConstructor = require(`./types/${typeFile}`);
+      this.addType(new TypeConstructor());
+    });
+  }
 
-/**
- * @param {Request} request
- * @param {boolean} verbose
- * @returns {Promise.<{documentBody: *, errorMessages:string[], valid: boolean}|KuzzleRequest>}
- */
-Validation.prototype.validationPromise = function validationValidationPromise (request, verbose) {
-  var
-    id = request.input.resource._id,
-    updateBodyClone,
-    isUpdate = false,
-    collectionSpec = {};
+  /**
+   * @param {Request} request
+   * @param {boolean} [verbose]
+   * @returns {Promise.<{documentBody: *, errorMessages:string[], valid: boolean}|KuzzleRequest>}
+   */
+  validationPromise(request, verbose = false) {
+    const
+      id = request.input.resource._id;
+    let
+      collectionSpec = {},
+      updateBodyClone,
+      isUpdate = false;
+
+    if (this.specification && this.specification[request.input.resource.index]
+      && this.specification[request.input.resource.index][request.input.resource.collection]) {
+      collectionSpec = this.specification[request.input.resource.index][request.input.resource.collection];
+    }
 
-  verbose = typeof verbose !== 'undefined' && verbose;
+    return new Bluebird(resolve => {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this function to use "return" consistently. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3801)